util-cgroups/tests/integration.rs

```diff
@@ -0,0 +1,892 @@
+            "Error, controler not expected found: {:?}",
+            hierarchy.available_controllers()
+        ))
+    }
+}
+
+/// Creates a directory within a given root path, initializes required files based on the
+/// specified cgroup version, and returns the path to the created directory.
+///
+/// # Example
+///
+/// ```rust
+/// use std::path::PathBuf;
+/// use plugin_cgroup::hierarchy::CgroupVersion;
+///
+/// let root = PathBuf::from("/tmp/test_cgroup");
+/// let folder = create_folder(&root, "my_cgroup", CgroupVersion::V2).unwrap();
+/// assert!(folder.exists());
+/// ```
+pub fn create_folder(root: &PathBuf, name: &str, hierarchy: CgroupHierarchy) -> Result<PathBuf, anyhow::Error> {
+    match hierarchy.version() {
+        CgroupVersion::V1 => {
+            // 1- Create the controller
+            // 2- Create the target
+            let dir_path = root.join(name);
+            fs::create_dir_all(&dir_path).unwrap();
+            assert!(is_accessible_dir(&dir_path).unwrap());
+            if hierarchy.available_controllers().contains(&String::from("cpuacct")) {
+                let _file_path = create_files_cgroupv1(&dir_path, hierarchy)?;
+            } else if hierarchy.available_controllers().contains(&String::from("memory")) {
+                let _file_path = create_files_cgroupv1(&dir_path, hierarchy)?;
+            } else {
+                return Err(anyhow!(
+                    "Error, controler not expected found: {:?}",
+                    hierarchy.available_controllers()
+                ));
+            }
+            Ok(dir_path)
+        }
+        CgroupVersion::V2 => {
+            // 1- Create the target
+            // 2- Create the controller
+            let dir_path = root.join(name);
+            fs::create_dir_all(&dir_path).unwrap();
+            assert!(is_accessible_dir(&dir_path).unwrap());
+            create_files_cgroupv2(&dir_path)?;
+            Ok(dir_path)
+        }
+    }
+}
+
+#[test]
+fn no_cgroup() {
+    let root_unwraped = tempdir().unwrap();
+    let root = root_unwraped.path().to_path_buf();
+    let version = CgroupVersion::V1;
+
+    //Creation of CgroupHierarchy
+    let hierarchy_cpu = CgroupHierarchy::manually_unchecked(root.clone(), version, vec![""]);
+
+    // Detection of Cgroup
+    let config = Config::default();
+    let f1 = callback(|cgroups| {
+        for cgroup in cgroups {
+            let s1: HashSet<_> = vec!["".to_string()].iter().cloned().collect();
+            let s2: HashSet<_> = cgroup.hierarchy().available_controllers().iter().cloned().collect();
+            let diff: Vec<_> = s1.difference(&s2).collect();
+            assert!(diff.len() == 0)
+        }
+        Ok(())
+    });
+    let f2 = callback(|_a| Ok(()));
+    let handler = ClosureCallbacks {
+        on_cgroups_created: f1,
+        on_cgroups_removed: f2,
+    };
+    let _detector = CgroupDetector::new(hierarchy_cpu, config, handler);
+}
+
+/// \\\\\\\\\\\\\\\\\
+/// Test for CgroupV1
```

Comment:
Since you have many integration tests, can you split them in modules? One for v1, one for v2 for instance.